distro/pkg/rpm/libyang.spec

```diff
@@ -34,21 +43,41 @@ written (and providing API) in C.
 
 %prep
 %autosetup -p1
-mkdir build
 
 %build
-cd build
-cmake \
-    -DCMAKE_INSTALL_PREFIX:PATH=%{_prefix} \
-    -DCMAKE_BUILD_TYPE:String="Release" \
-    -DCMAKE_C_FLAGS="${RPM_OPT_FLAGS}" \
-    -DCMAKE_CXX_FLAGS="${RPM_OPT_FLAGS}" \
-    ..
-make
+%cmake -DCMAKE_BUILD_TYPE=RELWITHDEBINFO
+%cmake_build
+
+%if %fedora == 34
+    %ifarch %{arm}
+        pushd armv7hl-redhat-linux-gnueabi
+    %else
+        pushd %{_host}
+    %endif
+%else
+    pushd redhat-linux-build
+%endif
+make doc
+popd
```

Comment:
I have just tested this on OpenSUSE and, as expected, this throws an error with invalid syntax. What worked for me was
```
%if 0%{?rhel}
    %if 0%{?fedora} == 34
        %ifarch %{arm}
            pushd armv7hl-redhat-linux-gnueabi
        %else
            pushd %{_host}
        %endif
    %else
        pushd redhat-linux-build
    %endif
%endif
make doc
%if 0%{?rhel}
    popd
%endif
```
and `%check` needed to be changed similarly.